src/overlays/actors/ovl_Elf_Msg2/z_elf_msg2.c

```diff
@@ -21,23 +31,162 @@ const ActorInit Elf_Msg2_InitVars = {
+        Flags_GetSwitch(globalCtx, this->actor.posRot.rot.y - 1)) {
+        LOG_STRING("共倒れ", "../z_elf_msg2.c", 171);
+        if ((this->actor.params >> 8 & 0x3F) != 0x3F) {
+            Flags_SetSwitch(globalCtx, (this->actor.params >> 8 & 0x3F));
+        }
+        Actor_Kill(&this->actor);
+        return 1;
+    }
+    // Checking a room clear flag:
+    else if ((this->actor.posRot.rot.y == -1) && Flags_GetClear(globalCtx, this->actor.room)) {
+        LOG_STRING("共倒れ２", "../z_elf_msg2.c", 182);
+        if ((this->actor.params >> 8 & 0x3F) != 0x3F) {
+            Flags_SetSwitch(globalCtx, (this->actor.params >> 8 & 0x3F));
+        }
+        Actor_Kill(&this->actor);
+        return 1;
+    } else if ((this->actor.params >> 8 & 0x3F) == 0x3F) {
+        return 0;
+    }
+    // Checking a switch or temp switch flag (from params):
+    else if (Flags_GetSwitch(globalCtx, (this->actor.params >> 8 & 0x3F))) {
+        LOG_STRING("共倒れ", "../z_elf_msg2.c", 192);
+        Actor_Kill(&this->actor);
+        return 1;
+    }
+    return 0;
+}
+
+void ElfMsg2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ElfMsg2* this = THIS;
+
+    osSyncPrintf(VT_FGCOL(CYAN) " Elf_Msg2_Actor_ct %04x\n\n" VT_RST, this->actor.params);
+    if (!ElfMsg2_KillCheck(this, globalCtx)) {
+        if (0 < this->actor.posRot.rot.x && this->actor.posRot.rot.x < 8) {
+            this->actor.unk_1F = (this->actor.posRot.rot.x - 1);
+        }
+        Actor_ProcessInitChain(thisx, sInitChain);
+        if (this->actor.posRot.rot.y >= 0x41) {
+            ElfMsg2_SetupAction(this, ElfMsg2_TextInit);
+        } else {
+            ElfMsg2_SetupAction(this, ElfMsg2_CheckForRead);
+            this->actor.flags = (this->actor.flags | 0x00040001);
+            this->actor.textId = ElfMsg2_GetMessageId(this);
+        }
+        this->actor.shape.rot.x = this->actor.shape.rot.y = this->actor.shape.rot.z = 0x0000;
+    }
+}
+
+void ElfMsg2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Elf_Msg2/ElfMsg2_Init.s")
+s32 ElfMsg2_GetMessageId(ElfMsg2* this) {
+    return (this->actor.params & 0xFF) + 0x100;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Elf_Msg2/ElfMsg2_Destroy.s")
+/**
+ * Runs while navi text is up. Handles the killing of the actor upon closing the text box, can also set a switch flag
+ * from params.
+ */
+void ElfMsg2_Read(ElfMsg2* this, GlobalContext* globalCtx) {
+    s32 switchFlag;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Elf_Msg2/func_809AD968.s")
+    // If the text box is closing this frame:
+    if (func_8002F334(&this->actor, globalCtx)) {
+        if (this->actor.posRot.rot.z != 1) {
+            Actor_Kill(&this->actor);
+            switchFlag = (this->actor.params >> 8) & 0x3F;
+            if (switchFlag != 0x3F) {
+                Flags_SetSwitch(globalCtx, switchFlag);
+            }
+            return;
```

Comment:
I'd remove the return here and put the `ElfMsg2_SetupAction(this, ElfMsg2_CheckForRead);` in an else block